src/components/tools/drawing-manager.js

```diff
@@ -6,42 +6,45 @@
  */
 AFRAME.registerComponent("drawing-manager", {
   init() {
-    this._onComponentInitialized = this._onComponentInitialized.bind(this);
-
     this.el.object3D.visible = false;
     this.drawingToPen = new Map();
   },
 
-  remove() {
-    if (this.drawingEl) {
-      this.drawingEl.removeEventListener("componentinitialized", this._onComponentInitialized);
-    }
-  },
-
-  _onComponentInitialized(e) {
-    if (e.detail.name == "networked-drawing") {
-      this.drawing = this.drawingEl.components["networked-drawing"];
-    }
-  },
-
-  createDrawing() {
-    if (!this.drawingEl) {
-      this.drawingEl = document.createElement("a-entity");
-      this.drawingEl.setAttribute("networked", "template: #interactable-drawing");
-      this.el.sceneEl.appendChild(this.drawingEl);
-
-      this.drawingEl.addEventListener("componentinitialized", this._onComponentInitialized);
-    }
+  destroyDrawing() {
+    this.drawingToPen.delete(this.drawing);
+    this.drawing = null;
+    this.drawingEl = null;
   },
 
   getDrawing(pen) {
     //TODO: future handling of multiple drawings
-    if (this.drawing && (!this.drawingToPen.has(this.drawing) || this.drawingToPen.get(this.drawing) === pen)) {
-      this.drawingToPen.set(this.drawing, pen);
-      return this.drawing;
-    }
-
-    return null;
+    return new Promise((resolve, reject) => {
```

Comment:
I know the TODO above was added before but -- any thoughts on how multiple drawings can be handled (in terms of technical requirements and UX)?

Reply:
Haven't given it too much thought yet because I'm not sure we even need it. It depends on how future functionality around manipulating/pinning drawings pans out.